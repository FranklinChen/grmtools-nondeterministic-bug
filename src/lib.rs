pub mod error;

use lrlex::lrlex_mod;
lrlex_mod!("term.l");

use lrpar::lrpar_mod;
lrpar_mod!("term.y");

#[cfg(test)]
mod tests {
    use super::*;
    use insta::assert_debug_snapshot;

    #[test]
    fn err_no_number() {
        let input = "(())";

        let lexerdef = term_l::lexerdef();
        let lexer = lexerdef.lexer(input);

        assert_debug_snapshot!(term_y::parse(&lexer));
    }
}
