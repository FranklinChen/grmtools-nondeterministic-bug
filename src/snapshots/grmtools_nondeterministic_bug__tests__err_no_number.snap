---
source: src/lib.rs
expression: "term_y::parse(&lexer)"
---
(
    Some(
        Err(
            LexemeError(
                DefaultLexeme {
                    start: 2,
                    len: 0,
                    faulty: true,
                    tok_id: 0,
                },
            ),
        ),
    ),
    [
        ParseError(
            ParseError {
                stidx: StIdx(
                    1,
                ),
                lexeme: DefaultLexeme {
                    start: 2,
                    len: 1,
                    faulty: false,
                    tok_id: 2,
                },
                repairs: [
                    [
                        Insert(
                            TIdx(
                                0,
                            ),
                        ),
                    ],
                ],
            },
        ),
    ],
)
